phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -757,6 +759,7 @@ synchronized public void close() throws SQLException {
             }
             
         } finally {
+            isClosing = false;
```

Comment:
It looks like we are bitten by some multi threading issue and we are trying to reduce the probability of that. @lokiore Is there something that i am missing?

Reply:
@mnpoonia this is being added as we don't have any flag unless all the layers below PhoenixConnection is already close and we needed something to know that Connection is being in process of closing.